src/setup-beam.js

```diff
@@ -17,21 +18,49 @@ async function main() {
   const elixirSpec = core.getInput('elixir-version', { required: false })
   const gleamSpec = core.getInput('gleam-version', { required: false })
   const rebar3Spec = core.getInput('rebar3-version', { required: false })
+  const shouldMixRebar = core.getInput('install-rebar', { required: false })
+  const shouldMixHex = core.getInput('install-hex', { required: false })
+  process.env.HEX_MIRROR = core.getInput('hexpm-mirror', { required: false })
 
   if (otpSpec !== 'false') {
-    await installOTP(otpSpec, osVersion)
-    const elixirInstalled = await maybeInstallElixir(elixirSpec, otpSpec)
+    const otpVersion = await getOTPVersion(otpSpec, osVersion)
+    const elixirVersion = await getElixirVersion(elixirSpec, otpVersion)
+    // No cache for Windows OTP, rebar3 and Gleam as they're downloaded from GitHub
+    const useCache =
+      isVersion(otpSpec) &&
+      (!elixirSpec || isVersion(elixirSpec)) &&
+      (elixirVersion || process.platform !== 'win32')
+    let hexCachedRestored = false
+    const cacheVersion = '1'
```

Comment:
What's the goal of setting this constant to only use it once, and on the line that follows?

Reply:
The cache contains  a set of files. Shall we need to invalidate it, the version should be incremented. It seems more obvious to me to have it on a separate line.

Reply:
To invalidate the caches shouldn't you just delete them, from (e.g. in our case) https://github.com/erlef/setup-beam/actions/caches?

Reply:
Consumers of the action wouldn't know when to invalidate the cache.

The other option would be to get a hash from the source code of the file as is often done with the cache action. I didn't find anything simple to do that without adding more dependencies though.

Reply:
It seems you did, after all, using just `crypto`.